backend/utils/management/commands/migrate_with_lock.py

```
@@ -0,0 +1,62 @@
+import zlib
+
+from django.core.management import BaseCommand, call_command
+from django.db import connections
+
+
+MIGRATION_LOCK_NAMESPACE = 0x54414C59  # "TALY", signed int32-safe.
+# ponytail: hardcoded; make it a flag/env var if a deploy ever needs to tune it.
+LOCK_TIMEOUT_SECONDS = 900
+
+
+def _signed_int32(value):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_signed_int32`

(ANN202)